daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto

```diff
@@ -127,3 +127,19 @@ message NodeRollback { // *since version 14*
 }
 
 // architecture-handbook-entry-end: Nodes
+
+message Versioned {
+  string version = 1;
+  bytes payload = 2;
+}
+
+message FatContractInstance {
+  bytes contract_id = 1;
+  com.daml.lf.value.Identifier template_id = 2;
+  bytes create_arg = 3;
+  KeyWithMaintainers contract_key_with_maintainers = 4;
+  repeated string non_maintainer_signatories = 5;
+  repeated string non_signatory_stakeholders = 6;
```

Comment:
```suggestion
  repeated string non_signatory_observers = 6;
```
unless there are other non-observer stakeholders?

Reply:
@andreaslochbihler-da, you proposed me the name  `non_signatory_stakeholders`. What  do you think of @meiersi-da suggestion ?

Reply:
This is the first time I see this name. How is it different than `observers`? Is it because `observers` can actually include some of the signatories?

Reply:
`observers` are the result of evaluating the `observer` function of a template. They may overlap with signatories.

I don't really care whether this is called observers or stakeholders. Given that it is an internal data definition for the Daml language, we can rename the field if necessary in the future without breaking backwards compatibility. Stakeholders is a native concept to the Daml engine code (create nodes contain the stakeholders, not the observers!), it feels natural to me to use `stakeholder` here. But I have no problem of putting `observer` here either.

Reply:
@remyhaemmerle-da I have not strong view on this. You be the judge :)

Reply:
`non_signatory_stakeholders` is an established field name, I am not in favor of introducing another